test/unit/date/format.js

```diff
@@ -464,6 +480,26 @@ QUnit.test( "should format timezone (z)", function( assert ) {
 	assert.dateFormat( date, "zzzz", cldr, "GMT+11:00" );
 });
 
+QUnit.test( "should format timezone (O)", function( assert ) {
+	var date = new FakeDate( 0 );
+	assert.dateFormat( date, "O", cldr, "GMT" );
+	assert.dateFormat( date, "OO", cldr, "GMT" );
+	assert.dateFormat( date, "OOO", cldr, "GMT" );
+	assert.dateFormat( date, "OOOO", cldr, "GMT" );
+
+	date = new FakeDate( -3 );
+	assert.dateFormat( date, "O", cldr, "GMT-3" );
+	assert.dateFormat( date, "OO", cldr, "GMT-3" );
+	assert.dateFormat( date, "OOO", cldr, "GMT-3" );
+	assert.dateFormat( date, "OOOO", cldr, "GMT-03:00" );
+
+	date = new FakeDate( 11 );
+	assert.dateFormat( date, "O", cldr, "GMT+11" );
+	assert.dateFormat( date, "OO", cldr, "GMT+11" );
+	assert.dateFormat( date, "OOO", cldr, "GMT+11" );
+	assert.dateFormat( date, "OOOO", cldr, "GMT+11:00" );
+});
```

Comment:
There's already a `"should format timezone (O)"` section above. Instead of duplicating it, let's simply incorporate the missing pieces there please.